ekss/adapters/vault/client.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Provides client side functionality for interaction with HashiCorp Vault"""
+
+
+from typing import Union
+
+import hvac
+
+
+class VaultClient:
+    """Wrapper around hvac client delegating actions"""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        url: str,
+        token: str,
+        namespace: str = "vault",
+        verify: Union[bool, str] = True,
+        timeout: int = 15,
+    ):
+        self.client = hvac.Client(
+            url=url,
+            namespace=namespace,
+            token=token,
+            verify=verify,
+            timeout=timeout,
```

Comment:
What is the reason for this wrapper? Currently, this class has no methods. If you only like to set change the defaults of the hvac.Client than it would be better to inherit from it and overload the __init__.

Reply:
I'd like some light abstractions like in auth-km-jobs, but I agree, should rather inherit from teh class.
Functionality is part of GDEV-1416